cli/azd/pkg/osutil/retry_strategy.go

```diff
@@ -0,0 +1,37 @@
+package osutil
+
+import (
+	"os"
+	"strings"
+	"time"
+)
+
+type RetryStrategy struct {
+	// The maximum number of retries before failing
+	MaxRetries uint64
+	// The time between each retry attempt
+	RetryBackoff time.Duration
+}
+
+// Creates a new retry strategy that also reduces the time for when running in test
```

Comment:
I found this behavior to be very surprising.  My hunch is that this speeds up negative testing because we don't sit trying to load a resource that will never be there.  I'm worried, however, that the cure is worse than the disease.  Are we going to start seeing eventual consistency issues in our tests that would have been handled by the actual retries?  The fact that we had to add logic here to opt out in some of our existing tests makes me really worried.

Reply:
This should only effect our unit tests and not our functional tests.  It should shave off literal minutes of our test pass.  Happy to discuss more or remove later if we see it become a problem.